src/main/java/com/dyescape/dyescapebot/moderation/PunishmentExpiryChecker.java

```diff
@@ -0,0 +1,60 @@
+package com.dyescape.dyescapebot.moderation;
+
+import com.dyescape.dyescapebot.entity.discord.ActivePunishment;
+import com.dyescape.dyescapebot.entity.discord.Warning;
+import com.dyescape.dyescapebot.repository.ModerationActivePunishmentRepository;
+import com.dyescape.dyescapebot.repository.ModerationWarningRepository;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.time.Instant;
+import java.util.List;
+
+@Component
+public class PunishmentExpiryChecker {
+
+    private final Moderation moderation;
+    private final ModerationActivePunishmentRepository activePunishmentRepository;
+    private final ModerationWarningRepository warningRepository;
+
+    public PunishmentExpiryChecker(Moderation moderation,
+                                   ModerationActivePunishmentRepository activePunishmentRepository,
+                                   ModerationWarningRepository warningRepository) {
+
+        this.moderation = moderation;
+        this.activePunishmentRepository = activePunishmentRepository;
+        this.warningRepository = warningRepository;
+    }
+
+    @Scheduled(fixedRate = 10000)
+    public void check() throws Exception {
+        List<ActivePunishment> expiredPunishments = this.activePunishmentRepository.findAllByEndBefore(Instant.now());
+        List<Warning> expiredWarnings = this.warningRepository.findAllByEndBefore(Instant.now());
+
+        for (ActivePunishment p : expiredPunishments) {
+            switch (p.getType()) {
+
+                case "MUTE":
```

Comment:
(Not sure) But maybe use an enum or constants?

Reply:
Was thinking of that indeed, but I think I ultimately chose to not do that to prevent database hassle in case we rename these types. Though we could just update the database really, so I'll make it an enum.